operator/pkg/remote/kyma.go

```diff
@@ -0,0 +1,184 @@
+		Name: fmt.Sprintf("%s.%s", operatorv1alpha1.KymaPlural, operatorv1alpha1.GroupVersion.Group),
+	}, &crd)
+	if err != nil {
+		return err
+	}
+	remoteCrd := v1extensions.CustomResourceDefinition{}
+	remoteCrd.Name = crd.Name
+	remoteCrd.Namespace = crd.Namespace
+	remoteCrd.Spec = crd.Spec
+	err = c.runtimeClient.Create(ctx, &remoteCrd)
+	if err != nil {
+		return err
+	}
+	return err
+}
+
+func (c *KymaSynchronizationContext) CreateOrFetchRemoteKyma(ctx context.Context) (*operatorv1alpha1.Kyma, error) {
+	kyma := c.controlPlaneKyma
+	recorder := adapter.RecorderFromContext(ctx)
+	remoteKyma := &operatorv1alpha1.Kyma{}
+	err := c.runtimeClient.Get(ctx, client.ObjectKeyFromObject(kyma), remoteKyma)
+
+	if meta.IsNoMatchError(err) {
+		recorder.Event(kyma, "Normal", err.Error(), "CRDs are missing in SKR and will be installed")
+		if err := c.CreateCRD(ctx); err != nil {
+			return nil, err
+		}
+		recorder.Event(kyma, "Normal", "CRDInstallation", "CRDs were installed to SKR")
+		// the NoMatch error we previously encountered is now fixed through the CRD installation
+		err = nil
+	}
+
+	if errors.IsNotFound(err) {
+		remoteKyma.Name = kyma.Name
+		remoteKyma.Namespace = kyma.Namespace
+		remoteKyma.Spec = *kyma.Spec.DeepCopy()
+		err = c.runtimeClient.Create(ctx, remoteKyma)
+		if err != nil {
+			recorder.Event(kyma, "Normal", "CRDInstallation", "CRDs were installed to SKR")
+			return nil, err
+		}
+	} else if err != nil {
+		recorder.Event(kyma, "Warning", err.Error(), "Client could not fetch remote Kyma")
+		return nil, err
+	}
+
+	return remoteKyma, err
+}
+
+func (c *KymaSynchronizationContext) SynchronizeRemoteKyma(ctx context.Context, remoteKyma *operatorv1alpha1.Kyma) (bool, error) {
+	kyma := c.controlPlaneKyma
+	recorder := adapter.RecorderFromContext(ctx)
+	// check finalizer
+	if !controllerutil.ContainsFinalizer(remoteKyma, labels.Finalizer) {
+		controllerutil.AddFinalizer(remoteKyma, labels.Finalizer)
+	}
+
+	if remoteKyma.Status.ObservedGeneration != remoteKyma.GetGeneration() {
+		kyma.Spec = remoteKyma.Spec
+		err := c.controlPlaneClient.Update(ctx, kyma)
+		if err != nil {
+			recorder.Event(remoteKyma, "Warning", err.Error(), "Client could not update Control Plane Kyma")
+			return true, err
+		}
+		remoteKyma.Status.ObservedGeneration = remoteKyma.GetGeneration()
+		err = c.runtimeClient.Status().Update(ctx, remoteKyma)
+		if err != nil {
+			return true, err
+		}
+	}
+
+	if remoteKyma.Status.State != kyma.Status.State {
+		remoteKyma.Status.State = kyma.Status.State
+		err := c.runtimeClient.Status().Update(ctx, remoteKyma)
+		if err != nil {
+			return true, err
+		}
+	}
+
+	lastSyncDate := time.Now().Format(time.RFC3339)
+	if remoteKyma.Annotations == nil {
+		remoteKyma.Annotations = make(map[string]string)
+	}
+	remoteKyma.Annotations[labels.LastSync] = lastSyncDate
+	err := c.runtimeClient.Update(ctx, remoteKyma)
+	if err != nil {
+		return true, err
+	}
+
+	return false, nil
```

Comment:
if remoteKyma is updated, subsequent updates will fail, since the cached object is the not the same anymore!